python/pyspark/sql/pandas/types.py

```diff
@@ -356,7 +389,27 @@ def _convert_map_items_to_dict(s: "PandasSeriesLike") -> "PandasSeriesLike":
     return s.apply(lambda m: None if m is None else {k: v for k, v in m})
 
 
-def _convert_dict_to_map_items(s: "PandasSeriesLike") -> "PandasSeriesLike":
+def _is_series_contain_timestamp(s):
+    """
+    checks whether the series contain Timstamp object
+    :param s: pd.series
+    :return:  True if the series contain timestamp object
+    """
+    from numpy import ndarray
+    import datetime
+    from pandas.api.types import is_datetime64tz_dtype, is_datetime64_dtype
+    if not s.empty and isinstance(s.dtype, object) and \
+            isinstance(s.iloc[0], list) and len(s.iloc[0]) > 0 and \
+            isinstance(s.iloc[0][0], datetime.datetime):
```

Comment:
I don't think you can rely on checking the first value to determine the data type

Reply:
Agree , However for array with pyarrow , I m checking the dtype for the array. 
But I think as suggested by you for pyarrow , it would be better to use pyarrow for type checking